osgp/platform/osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/AdHocManagementService.java

```diff
@@ -126,7 +128,9 @@ public void handleGetLightSensorStatusResponse(
     response.setOsgpException(exception);
     response.setResult(deviceResult);
     if (lightSensorStatusDto != null) {
-      response.setLightSensorStatus(new LightSensorStatus(lightSensorStatusDto.isOn()));
+      response.setLightSensorStatus(
+          new LightSensorStatus(
+              LightSensorStatusType.valueOf(lightSensorStatusDto.getStatus().toString())));
```

Comment:
When used in a functional way (and not only informational) it is probably better not to rely on `toString()` but use something more specific. For enums I think either `name()` or a custom property would be more robust choices.